Convert MatchResult to LP format

NguyenLan94 opened the pull request.

NguyenLan94 committed:
Convert MatchResult to LP format

NguyenLan94 committed:
Fix bug wrong winner number

justthieenj commented on src/main/java/io/github/justthieenj/dotafetchmatch/scripts/Main.java:
```diff
@@ -24,14 +24,15 @@ private static void execute() {
         sleep(1000);
         var matchResult = dotabuff().getMatchResult();
         System.out.println(matchResult.beautify());
+        System.out.println(convertToLPMatch(matchResult, matchResult.getTeam1(), matchResult.getTeam2()));
```
no need 2nd and 3rd String params, the MatchResult data already cover it

justthieenj commented on src/main/java/io/github/justthieenj/dotafetchmatch/dataobject/MatchResult.java:
```diff
@@ -36,4 +38,46 @@ public String beautify() {
                """.formatted(matchId, map, sideWinner, teamWinner, length, radiantTeam, radiantBans, radiantPicks, direTeam, direBans, direPicks);
     }
+
+    public String convertTeam1Result(List<String> team1picks, List<String> team1bans) {
```
the convertTeam1Result() and convertTeam2Result() are identical, refactor it and give it a 3rd params (such as `int teamNumber`)

`convertTeamResult(int teamNumber, List<String> picks, List<String> bans)`

justthieenj replied:
also use loop instead of go through `list.get(index)` like this

justthieenj commented on src/main/java/io/github/justthieenj/dotafetchmatch/page/Dotabuff.java:
```diff
@@ -28,6 +28,11 @@ private String getTeam(TeamSide side) {
         return element.getText();
     }
 
+    private String getTeam(int teamNumber) {
```
0 and 1 arent really a `teamNumber`, refactor this method or change param name

justthieenj commented on src/main/java/io/github/justthieenj/dotafetchmatch/dataobject/MatchResult.java:
```diff
@@ -36,4 +38,46 @@ public String beautify() {
+    private String convertLengthFormat(String length) {
+        var split = length.split(":");
+        int minute = Integer.parseInt(split[0]) * 60 + Integer.parseInt(split[1]);
+        return minute + "m" + split[2] + "s";
```
handle hour value.
refer to this match:
https://www.dotabuff.com/matches/7629175264
https://liquipedia.net/dota2/index.php?title=DreamLeague/Season_22&action=edit&section=9
Dotabuff: `1:02:55` -> LP value should be `62m55s`

justthieenj commented on src/main/java/io/github/justthieenj/dotafetchmatch/dataobject/MatchResult.java:
```diff
@@ -36,4 +38,46 @@ public String beautify() {
+                convertLengthFormat(length), sideWinner == TeamSide.radiant ? 1 : 2);
+    }
+    private String convertLengthFormat(String length) {
+        var split = length.split(":");
```
this is a `String[]`. update variable name to be more clear

NguyenLan94 committed:
- Refactor MatchResult.convertTeamResult() function to use loop
- Update variable name in MatchResult.convertLengthFormat() function
- Optimize Main.convertToLPMatch() function
- Refactor Dotabuff.getTeam() function
- Fix bug wrong winner number

justthieenj committed:
wip refactor

justthieenj committed:
create TeamData object to store data

justthieenj committed:
run headless
